maven-scm-providers/maven-scm-providers-git/maven-scm-provider-jgit/src/main/java/org/apache/maven/scm/provider/git/jgit/command/checkin/JGitCheckInCommand.java

```diff
@@ -139,7 +141,19 @@ protected CheckInScmResult executeCheckInCommand(
                 }
                 RefSpec refSpec = new RefSpec(Constants.R_HEADS + branch + ":" + Constants.R_HEADS + branch);
                 logger.info("push changes to remote... " + refSpec);
-                JGitUtils.push(git, (GitScmProviderRepository) repo, refSpec);
+                Iterable<PushResult> pushResultList = JGitUtils.push(git, (GitScmProviderRepository) repo, refSpec);
+
+                for (PushResult pushResult : pushResultList) {
+                    for (RemoteRefUpdate remoteRefUpdate : pushResult.getRemoteUpdates()) {
+                        if (remoteRefUpdate.getStatus() != RemoteRefUpdate.Status.OK) {
```

Comment:
I think up2date is no error condition either: https://archive.eclipse.org/jgit/docs/jgit-2.0.0.201206130900-r/apidocs/org/eclipse/jgit/transport/RemoteRefUpdate.Status.html#UP_TO_DATE